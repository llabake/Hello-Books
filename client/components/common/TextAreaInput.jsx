import React from 'react';
import PropTypes from 'prop-types';


const TextAreaInput = ({ id, icon, placeholder, errors, name, value, onChange, }) => {
  return (
    <div className="row">
      <div className="input-field col s12">
        <i className="material-icons prefix">{icon}</i>
        <textarea 
        id={id} 
        className="materialize-textarea validate"
        placeholder={placeholder}
        value={value}
        name ={name}
        onChange={onChange}
        />
        {errors && errors.length ?
        errors.map((error, i) => { return (
        <div key={i} className= 'red-text left-align' style={{ 'fontSize':'medium', 'marginLeft': 46}}>
          {/* <i className="material-icons">error_outline</i> */}
          {error}
        </div>
        )}) : null }
      </div>
    </div>
  )

}


TextAreaInput.propTypes = {
  id: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  value: PropTypes.string,
  name: PropTypes.string,
  errors: PropTypes.array,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};


export default TextAreaInput;
