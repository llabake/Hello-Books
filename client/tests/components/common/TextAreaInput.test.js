import React from 'react';

import TextAreaInput from '../../../components/common/TextAreaInput';

test('Text area input field component renders correctly', () => {
  const props = {
    id: '',
    type: '',
    icon: '',
    placeholder: '',
    name: '',
    errors: [],
    value: '',
    onChange: jest.fn(),
  }
  const wrapper = mount(<TextAreaInput {...props} />);
  expect(wrapper).toBeDefined();
  expect(wrapper).toMatchSnapshot();
})

test('Text input field component show error when they occur', () => {
  const props = {
    id: '',
    type: '',
    icon: '',
    placeholder: '',
    name: '',
    errors: [
      'content is required'
    ],
    value: '',
    onChange: jest.fn(),
    onBlur: jest.fn()
  }
  const wrapper = mount(<TextAreaInput {...props} />);
  expect(wrapper).toBeDefined();
  expect(wrapper.find('div.red-text').exists()).toBeTruthy();
  expect(wrapper).toMatchSnapshot();
})