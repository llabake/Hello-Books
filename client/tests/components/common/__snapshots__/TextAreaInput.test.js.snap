// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text area input field component renders correctly 1`] = `
<TextAreaInput
  errors={Array []}
  icon=""
  id=""
  name=""
  onChange={[MockFunction]}
  placeholder=""
  type=""
  value=""
>
  <div
    className="row"
  >
    <div
      className="input-field col s12"
    >
      <i
        className="material-icons prefix"
      />
      <textarea
        className="materialize-textarea validate"
        id=""
        name=""
        onChange={[MockFunction]}
        placeholder=""
        value=""
      />
    </div>
  </div>
</TextAreaInput>
`;

exports[`Text input field component show error when they occur 1`] = `
<TextAreaInput
  errors={
    Array [
      "content is required",
    ]
  }
  icon=""
  id=""
  name=""
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  placeholder=""
  type=""
  value=""
>
  <div
    className="row"
  >
    <div
      className="input-field col s12"
    >
      <i
        className="material-icons prefix"
      />
      <textarea
        className="materialize-textarea validate"
        id=""
        name=""
        onChange={[MockFunction]}
        placeholder=""
        value=""
      />
      <div
        className="red-text left-align"
        key="0"
        style={
          Object {
            "fontSize": "medium",
            "marginLeft": 46,
          }
        }
      >
        content is required
      </div>
    </div>
  </div>
</TextAreaInput>
`;
